pyDeltaRCM/init_tools.py

```diff
@@ -85,6 +85,9 @@ def import_files(self):
         for k, v in list(input_file_vars.items()):
             setattr(self, k, v)
 
+        if self.seed >= 0:
+            np.random.seed(self.seed)
```

Comment:
I'm not sure about this. What if some goofball sets the seed to be a negative number? I could easily see someone doing that, in which case they'll get confusing behavior. Can we do something instead where the default is `None` and then this section of code does the logic parsing for that, specifying it to any random number if the person doesn't specify, and if they do specify, then use that one.

Reply:
@ericbarefoot yeah, I agree the logic here is bad. I think ideal would be `seed: None` in the yaml, and then
```python
if self.seed:
    np.random.seed(self.seed)
```
It could be that simple, if you could make the default `None`... But, if you specify the type of seed in `default.yml` to be `int`, then `seed: None` in the config yaml, you get back a type error. 

I couldn't readily think of a clean way to do it, and `-1` is often used as a special case in python args, so I went with it. I'll look at this some more.

Reply:
err nvm that would evaluate to `False` for seed=0. Yours above would work though, I think, if `None` were `int`...

Reply:
@ericbarefoot yah I'm scratching my head over here for ways to do this.

I can get it working by specifying the default value as `[null, 0]` and then ignoring your yaml key `type:` and just using the type returned from the parsing of the yaml key `default:`, but this is really unreadable in the default file, and I think the default of -1 is much more intuitive. We can just specify in the docs that only positive integers are acceptable. 

I don't want to do something like `if type(user_dict[oo]) is expected_type or user_dict[oo] is None` because that would open any key to be None too, which I think would be bad.

Happy to take suggestions on how to make it work with taking `null` or an integer.

Reply:
Also:

```python
 import numpy as np

np.random.seed(-10)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "mtrand.pyx", line 244, in numpy.random.mtrand.RandomState.seed
  File "_mt19937.pyx", line 166, in numpy.random._mt19937.MT19937._legacy_seeding
  File "_mt19937.pyx", line 180, in numpy.random._mt19937.MT19937._legacy_seeding
ValueError: Seed must be between 0 and 2**32 - 1
```

So it wouldn't even work if user specified a negative value.

Reply:
something we could think about is a more flexible parser. This also resolves another issue, that we call `eval()` on arbitrary yaml; I think pyyaml does this already anyway, but avoiding calls to `eval` is always good.

We could implement something like `numeric` as a default, that allows floats or ints, or lists `[int, None]` that allows either.

Reply:
> So it wouldn't even work if user specified a negative value.

That's good to know. I figured that the seed setting was a little more flexible. Since this is the case, I don't mind the -1 default actually. I was just worried about people throwing in random garbage accidentally. If numpy throws an error, then great. 

> but avoiding calls to eval is always good.

For sure. I couldn't think of a better way to test for type, and if there's a better way, then I'm happy to go that route.